### DISCUSSION TITLE
Git plugin cannot be disabled and slows down prompt making it unusable

### DISCUSSION THREAD

**Discussion by lucian303** [ohmyzsh/ohmyzsh#9849] [Help] [answered]
The prompt is very, very, very slow and it still shows the git branch even after removing all plugins in the plugin config and rm -rf ing the .oh-my-zsh/plugins dir. What does one have to do to get rid of this and make this to work again short of removing the entire .oh-my-zsh folder? Why is this imposed in such a way that it cannot be stopped? 

This happens after a fresh install, w/ a fresh config, with or without 'git' in the plugins line. Surely this can't be intended behavior?

It also happens for prompts that show the git branch but not at least for xiong-chiamiov or others (I assume that don't). 

**To Reproduce**
cd to a big git repo

**Expected behavior**
It shouldn't run any git stuff. It shouldn't display the git branch name. It shouldn't do any of this shit when it's disabled.

**Desktop (please complete the following information):**

 - OS / Distro: Debian 4.9.0
 - Latest ohmyzsh update?: yes
 - ZSH Version: 5.7.1
 - Terminal emulator: iTerm2

**Comment by mcornella** [comment, marked as answer]
That is the `git_prompt_info` / `git_prompt_status` / `parse_git_dirty` functions. You can disable them per repository with one of these run inside the git repository:
```zsh
git config oh-my-zsh.hide-info 1   # git_prompt_info
git config oh-my-zsh.hide-status 1 # git_prompt_status
git config oh-my-zsh.hide-dirty 1  # parse_git_dirty
```
You can also disable them entirely by either (1) redefining the function with an empty body in your zshrc file (after OMZ is sourced), or (2) running the commands above with a `--global` flag.

1. 
  ```zsh
  git_prompt_info(){} # this overwrites the git_prompt_info so it does nothing
  ```
2. 
  ```zsh
  git config --global oh-my-zsh.hide-info 1   # git_prompt_info
  git config --global oh-my-zsh.hide-status 1 # git_prompt_status
  git config --global oh-my-zsh.hide-dirty 1  # parse_git_dirty
  ```

In our experience `git_prompt_status` and `parse_git_dirty` are the ones that cause performance issues, so you can disable them and keep the `git_prompt_info` function to have branch info with none of the performance drawbacks.

**Reply by mcornella** [reply]
Addendum: we recently (a couple months ago) deployed a new async git prompt which should take care of this. It's still not perfect so check out https://github.com/ohmyzsh/ohmyzsh#disable-async-git-prompt if you're having issues.

**Comment by imxys** [comment]
Having the same issue, I find the source code by filtering `git_prompt_info`:
```bash
plugins/gitfast/gitfast.plugin.zsh
8:function git_prompt_info() {

lib/git.zsh
12:function git_prompt_info() {
```
Does the gitfast plugin overwrite the origin `git_prompt_info` and ignore `oh-my-zsh.hide-info`?

**Reply by imxys** [reply]
Oh sorry I made a mistake, my zsh runs git through powerlevel10k theme...
